bionic/deriver.py

```diff
@@ -367,8 +369,32 @@ class Bootstrap:
 
     persistent_cache = attr.ib()
     versioning_policy = attr.ib()
-    process_executor = attr.ib()
     process_manager = attr.ib()
+    process_executor = attr.ib()
+
+
+class SynchronizedSet:
+    """
+    A set where add() operation is thread-safe.
```

Comment:
I can see the confusion by calling this a set. I do think saying `add` is thread-safe is accurate since the definition of add has changed (returns true if added a new key). I'll reword it.